Microsoft.Toolkit.Uwp.UI.Animations/CompositionAnimations/Animations/TypedAnimationBase.cs

```diff
@@ -0,0 +1,165 @@
+// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+
+namespace Microsoft.Toolkit.Uwp.UI.Animations
+{
+    /// <summary>
+    /// A generic class extending <see cref="AnimationBase"/> to provide common implementation for most animations
+    /// </summary>
+    /// <typeparam name="T">Type of <see cref="TypedKeyFrame{U}"/> to use</typeparam>
+    /// <typeparam name="U">Type of value being animated.</typeparam>
+    public abstract class TypedAnimationBase<T, U> : AnimationBase
+        where T : TypedKeyFrame<U>, new()
+    {
+        private T fromKeyFrame;
+        private T toKeyFrame;
+
+        /// <summary>
+        /// Identifies the <see cref="From"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty FromProperty =
+            DependencyProperty.Register("From", typeof(U), typeof(TypedAnimationBase<T, U>), new PropertyMetadata(GetDefaultValue(), OnAnimationPropertyChanged));
```

Comment:
`From` should use `nameof(From)` (same for DPs below)

Reply:
Will fix

Reply:
Done